v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/uniformsplitter/range/BoundarySplitterFactory.java

```diff
@@ -263,4 +268,29 @@ protected static Timestamp instantToTimestamp(Instant instant) {
   private static Timestamp splitTimestamps(Timestamp start, Timestamp end) {
     return instantToTimestamp(splitInstants(timeStampToInstant(start), timeStampToInstant(end)));
   }
+
+  private static Float splitFloats(Float start, Float end) {
+      if (start == null && end == null) {
+          return null;
+      }
+      if (start == null) {
+          start = -Float.MAX_VALUE;
+      }
+      if (end == null) {
+          end = Float.MAX_VALUE;
+      }
+
+      // Calculate overflow safe mid-point
+
+      // If signs are different, simple addition is safe from overflow
+      // because the values cancel each other out towards zero.
+      if ((start < 0 && end > 0) || (start > 0 && end < 0)) {
```

Comment:
For float and double we also need to enhance[ isSplittable](https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/a73e0f07a719c668396e09de5195e7a9a0b756ce/v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/uniformsplitter/range/Boundary.java#L147) to stop the splitting at appropriate precision.